pkg/dockerutil/container_host.go

```
@@ -3,30 +3,95 @@ package dockerutil
 import (
 	"fmt"
 	"net"
+	"os/exec"
 	"runtime"
+	"strings"
 
 	"gopkg.in/inconshreveable/log15.v2"
 )
 
+func ifaceIPAddr(ifaceName string) (ip string, err error) {
+	iface, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		return "", err
+	}
+	addrs, err := iface.Addrs()
+	if err != nil {
+		return "", err
+	}
+	for _, addr := range addrs {
+		if ipn, ok := addr.(*net.IPNet); ok {
+			return ipn.IP.String(), nil
+		}
+	}
+	return "", nil
```

Hmm.. I guess we could return an error, but unless we return the more verbose error:

```Go
fmt.Errorf("unable to determine Docker container host address (as seen by containers), using interface %s", iface)
```

then I don't think it would help clarify the calling code, would it? Maybe that verbose error (+ `log.Crit`) would be ok here? what do you think?